docs/semgrep-code/semgrep-assistant-code.md

```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
 * Semgrep Inc. sends relevant lines of code to OpenAI's API, where currently, the "relevant lines of code" means lines that are part of the Semgrep finding, plus 10 lines of context on each side. Semgrep Inc. is likely to expand this, potentially to the entire file as we learn how to pass more useful context.
 * Semgrep Inc stores and retains these code snippets for up to 6 months. Semgrep Inc. will update you with at least a 30-day notice, if we make any changes to the retention policy.
 * Semgrep Inc is a paying customer of OpenAI and has a Data Protection Agreement signed with them (provided upon request by contacting [support@semgrep.com](mailto:support@semgrep.com). The code snippets we upload will be persisted by OpenAI temporarily, following their data usage policies at [Open AI API data Usage Policies](https://openai.com/policies/api-data-usage-policies).
 * For more details, see the [Semgrep Assistant FAQ](https://get.semgrep.dev/assistant).
 
+## Required GitHub or GitLab repository permissions
+
+Semgrep Assistant requires the same permissions that Semgrep needs to integrate with GitHub or GitLab, such as permissions that allow Semgrep to run CI jobs and post comments to PRs or MRs. See [<i class="fa-regular fa-file-lines"></i> Requested permissions for GitHub and GitLab](/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab) for a list of permissions.
+
+Assistant extends normal Semgrep functionality by providing contextually aware AI-generated suggestions. In order to build that context, it requires extra permissions in GitHub and GitLab.
+
+<Tabs
+    defaultValue="github"
+    values={[
+    {label: 'GitHub', value: 'github'},
+    {label: 'GitLab', value: 'gitlab'},
+    ]}
+>
+
+<TabItem value='github'>
+
+Semgrep Assistant requires [read access to your code in GitHub](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28). This is done through a private Semgrep GitHub app that you install during Assistant setup. This private Semgrep GitHub app:
+
+* Is fully under your control so you can revoke access or specific permissions at any time by visiting **Settings > Applications** in GitHub.
+* Only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* Can be configured to limit its scope to specific repositories. You do not need to give read access to all repositories in your GitHub organization. 
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html). 
```

Comment:
I noticed when I was going over this again that we say "self-managed GitLab" here but I think I've usually seen us say "GitLab Self-Managed". However, we do use 'self-hosted' as a prefixing generic in a few places so...whatever makes the most sense!

Reply:
"GitLab or self-managed GitLab" does sound a bit weird. To make it flow a little better: "Semgrep Assistant requires the API scope to run in both cloud and self-managed GitLab instances."